pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -694,7 +696,9 @@ func (service *SecurityPolicyService) deleteSecurityPolicy(sp types.UID) error {
 	finalSecurityPolicyCopy := *nsxSecurityPolicy
 	finalSecurityPolicyCopy.Rules = nsxSecurityPolicy.Rules
 
-	infraSecurityPolicy, err := service.WrapHierarchySecurityPolicy(nsxSecurityPolicy, *nsxGroups)
+	// Separate the security policy and rules deletion from groups deletion in case group deletion may block security policy/rule deletion in HAPI
+	nsxGroupsEmpty := &([]model.Group{})
```

Comment:
for passing an empty group:
```suggestion
	nsxEmptyGroups:= []model.Group{}
```

Reply:
fixed